app/controllers/email_confirmations_controller.rb

```diff
@@ -1,5 +1,7 @@
 class EmailConfirmationsController < ApplicationController
   before_action :redirect_to_signin, unless: :signed_in?, only: :unconfirmed
+  before_action :redirect_to_new_mfa, if: :mfa_required_not_yet_enabled?
+  before_action :redirect_to_settings_strong_mfa_required, if: :mfa_required_weak_level_enabled?
```

Comment:
redirect to sign in only occurs for the `unconfirmed` action. It doesn't make sense to redirect to mfa pages for `update` and `mfa_update` because the user isn't in a valid session (`validate_confirmation_token` updates the user using the confirmation token but the user isn't signed in). I guess `mfa_required_not_yet_enabled?` returns false anyways because `current_user` is nil. If we want to block this action, we should send an error back.

Thinking about this more... do email confirmations only occur when a user first signs up for an account? If so, I don't think it makes sense to have these checks because they wouldn't be possible for them to be a part of the mfa req cohort.

I see that you're only testing the new action.. when is that action triggered?

Reply:
new_email_confirmations_path is used for the resend confirmation. I kind of just picked the top action for every file to test, although I could see why we would want to be more specific with the tests.  Regarding it being too early to force mfa, is it possible for an account without a confirmed email to become owner of a gem?

Reply:
> Regarding it being too early to force mfa, is it possible for an account without a confirmed email to become owner of a gem?

Good question, my gut said no... you would need to confirm your email first in order to be an owner of something. I was wrong though, so we can keep this in 👍  

If someone adds you as an owner of a gem, you will receive an email confirmation for the new ownership. You can be added by confirming that email confirmation, even if you didn't confirm your new account email confirmation.